draft-ietf-gnap-core-protocol.md

```diff
@@ -1899,54 +1688,85 @@ properties.
+    access token. If provided, the client instance MAY manage its access
+    token as described in {{token-management}}. This management
+    URI is a function of the AS and is separate from the RS
+    the client instance is requesting access to.
+    This URI MUST NOT include the
+    access token value and SHOULD be different for each access
+    token issued in a request.
+
+access (array of objects/strings)
 : RECOMMENDED. A description of the rights
-              associated with this access token, as defined in 
-              {{request-resource-single}}. If included, this MUST reflect the rights
-              associated with the issued access token. These rights MAY vary
-              from what was requested by the client instance.
+    associated with this access token, as defined in 
+    {{resource-access-rights}}. If included, this MUST reflect the rights
+    associated with the issued access token. These rights MAY vary
+    from what was requested by the client instance.
 
 expires_in (integer)
 : OPTIONAL. The number of seconds in
-              which the access will expire. The client instance MUST NOT use the access
-              token past this time. An RS MUST NOT accept an access token
-              past this time. Note that the access token MAY be revoked by the
-              AS or RS at any point prior to its expiration.
-
-key (object / string / boolean)
-: REQUIRED. The key that the token is bound to. If the boolean value `true` is used,
-              the token is bound to the [key used by the client instance](#request-client) in its request 
-              for access. If the boolean value `false` is used,
-              the token is a bearer token with no key bound to it.
-              Otherwise, the key MUST be an object or string in a format
-              described in {{key-format}}, describing a public key to which the
-              client instance can use the associated private key. The client instance MUST be able to
-              dereference or process the key information in order to be able
-              to sign the request.
-
-The following non-normative example shows a single bearer token with a management
-URL that has access to three described resources.
+    which the access will expire. The client instance MUST NOT use the access
+    token past this time. An RS MUST NOT accept an access token
+    past this time. Note that the access token MAY be revoked by the
+    AS or RS at any point prior to its expiration.
+
+key (object / string)
+: OPTIONAL. The key that the token is bound to, if different from the
+    client instance's presented key. The key MUST be an object or string in a format
+    described in {{key-format}}, describing a public key to which the
+    client instance can use the associated private key. The client instance MUST be able to
+    dereference or process the key information in order to be able
+    to sign the request.
+
+durable (boolean)
```

Comment:
I don't see how this is useful, IOW how the client's code would differ depending on this setting. In either case the access token might suddenly expire, and the client needs to deal with that.

Reply:
Token rotation is being tracked on #161 #101 #96 and a number of other issues. This function was previously an input flag from the client instance, but the editors are here framing it as an output state flag instead because there wasn't a lot of support for having the client signal this going in.